src/Mod/PartDesign/Gui/TaskHoleParameters.cpp

```diff
@@ -388,55 +382,92 @@ void TaskHoleParameters::holeCutTypeChanged(int index)
+    bool isCounterdrill = (holeCutTypeString == "Counterdrill");
+    bool includeAngle = hole->DrillForDepth.getValue();
+    bool isNotCut = holeCutTypeString == "None";
+
     ui->labelHoleCutDiameter->setHidden(isNotCut);
     ui->labelHoleCutDepth->setHidden(isNotCut);
-    ui->labelHoleCutCountersinkAngle->setHidden(isNotCut);
     ui->HoleCutDiameter->setHidden(isNotCut);
     ui->HoleCutDepth->setHidden(isNotCut);
-    ui->HoleCutCountersinkAngle->setHidden(isNotCut);
-    ui->HoleCutCustomValues->setHidden(isNotCut);
 
-    if (HoleCutTypeString == "None" || HoleCutTypeString == "Counterbore"
-        || HoleCutTypeString == "Countersink" || HoleCutTypeString == "Counterdrill") {
-        ui->HoleCutCustomValues->setEnabled(false);
-        if (HoleCutTypeString == "None") {
-            ui->HoleCutDiameter->setEnabled(false);
-            ui->HoleCutDepth->setEnabled(false);
-            ui->HoleCutCountersinkAngle->setEnabled(false);
-        }
-        if (HoleCutTypeString == "Counterbore") {
-            ui->HoleCutCountersinkAngle->setEnabled(false);
-        }
-        if (HoleCutTypeString == "Countersink") {
-            ui->HoleCutCountersinkAngle->setEnabled(true);
-        }
+    std::string baseFileName;
+    if (isCounterbore) {
+        baseFileName = "hole_counterbore";
+        ui->HoleCutCountersinkAngle->setVisible(false);
+        ui->labelHoleCutCountersinkAngle->setVisible(false);
     }
-    else {  // screw definition
-        // we can have the case that we have no normed values
-        // in this case HoleCutCustomValues is read-only AND true
-        if (ui->HoleCutCustomValues->isChecked()) {
-            ui->HoleCutDiameter->setEnabled(true);
-            ui->HoleCutDepth->setEnabled(true);
-            if (!hole->HoleCutCountersinkAngle.isReadOnly()) {
-                ui->HoleCutCountersinkAngle->setEnabled(true);
-            }
-        }
-        else {
-            ui->HoleCutCustomValues->setEnabled(true);
-            ui->HoleCutDiameter->setEnabled(false);
-            ui->HoleCutDepth->setEnabled(false);
-            ui->HoleCutCountersinkAngle->setEnabled(false);
-        }
+    else if (isCountersink) {
+        baseFileName = "hole_countersink";
+        ui->HoleCutCountersinkAngle->setVisible(true);
+        ui->labelHoleCutCountersinkAngle->setVisible(true);
+    }
+    else if (isCounterdrill) {
+        baseFileName = "hole_counterdrill";
+        ui->HoleCutCountersinkAngle->setVisible(true);
+        ui->labelHoleCutCountersinkAngle->setVisible(true);
+    }
+    else {
+        baseFileName = "hole_none";
+        ui->HoleCutCountersinkAngle->setVisible(false);
+        ui->labelHoleCutCountersinkAngle->setVisible(false);
+    }
+
+    if (isAngled) {
+        baseFileName += includeAngle ? "_angled_included" : "_angled";
+    } else {
+        baseFileName += "_flat";
     }
+
+    ui->cutDiagram->setSvg(
+        QString::fromUtf8((":images/" + baseFileName + ".svg").c_str())
+    );

```

Comment:
This can be simplified to:
```suggestion
    ui->cutDiagram->setSvg(QString::fromStdString(":images/" + baseFileName + ".svg"));
```